tests/batchcheck/batchcheck_test.go

```diff
@@ -0,0 +1,47 @@
+package batchcheck
+
+import (
+	"testing"
+	"time"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"go.uber.org/goleak"
+
+	"github.com/openfga/openfga/internal/server/config"
+	"github.com/openfga/openfga/pkg/testutils"
+	"github.com/openfga/openfga/tests"
+)
+
+func TestBatchCheckMemory(t *testing.T) {
+	testRunAll(t, "memory")
+}
+
+func TestCheckPostgres(t *testing.T) {
+	testRunAll(t, "postgres")
+}
+
+func TestCheckMySQL(t *testing.T) {
+	testRunAll(t, "mysql")
+}
+
+func TestCheckSQLite(t *testing.T) {
+	testRunAll(t, "sqlite")
+}
```

Comment:
The reason why we originally wrote these tests for Check, ListObjects, ListUsers is because the APIs are complex and there are thousands of combinations of inputs that we wanted to account for.

- That's not the case for BatchCheck, which doesn't have complex business logic.
- If Check breaks while developing something, the test will also break for BatchCheck too, which is annoying/undesirable.
- Consider how much time this adds to the test suite:

```
 ok  	github.com/openfga/openfga/tests/batchcheck	66.842s
```

Given the above, I was wondering whether what we need are unit tests and not integration tests.

Reply:
I agree. These integration tests are mostly just testing that the check responses come back grouped and formatted properly, which is also covered in the unit tests. At the very least, we don't need them for all 4 datastores, but I actually don't think we need them at all; we're getting the coverage elsewhere